File: logger/src/main/java/ru/surfstudio/android/logger/remote_logging_strategies/RemoteLoggingStrategy.java
package ru.surfstudio.android.logger.remote_logging_strategies;

public interface RemoteLoggingStrategy {

    void setUser(String id, String username, String email);

    void clearUser();

    void logError(Throwable e);

    void logMessage(String message);

    void logKeyValue(String key, String value);
}

File: logger/lib-logger/src/main/java/ru/surfstudio/android/logger/remote_logging_strategies/RemoteLoggingStrategy.kt
package ru.surfstudio.android.logger.remote_logging_strategies

/**
 * Интерфейс для стратегий логгирования на удаленный сервер
 */
interface RemoteLoggingStrategy {

    fun setUser(id: String?, username: String?, email: String?)

    fun clearUser()

    fun logError(e: Throwable?)

    fun logMessage(message: String?)

    fun logKeyValue(key: String?, value: String?)
}
